### ISSUE TITLE
Support new includeIf.<condition>.path directive

### ISSUE THREAD

**Issue by kballard** [libgit2/libgit2#4246]
Git 2.13 added support for a new `includeIf.<condition>.path` directive, to conditionally include a config file depending on the location of the gitdir. It appears that libgit2 doesn't yet support this, but it should. In particular I really want GitUp to be able to support this and it relies on libgit2 (see git-up/GitUp#315).

**Comment by pks-t** [comment]
Incidentally, I've just started working on this feature in #4244. So you can expect it to appear after v0.26.0 has been released.

**Comment by andreineculau** [comment]
@pks-t just want to confirm - this issue has a PR https://github.com/libgit2/libgit2/pull/4244 from May 30, which is held up by another PR https://github.com/libgit2/libgit2/pull/4250 from May 30 ?
Those PRs have not been touched since a month ago (June 6).

Correct? What is needed to move forward?

**Comment by pks-t** [comment]
@andreineculau Yes, there is some work in progress to support conditional includes, and they'll most likely be part of the next release. Until now, things have not been merged due to the previous release v0.26.0, which put us in a feature freeze, and afterwards there ware some more pressing issues which I wanted to tackle before getting the rest merged. This is mostly stuff regarding our build infrastructure, which I want to get merged asap in the current release cycle.

Also, there is the problem of lack of time. We always peer review PRs before having them merged. So if you want to speed this up you're always welcome to review the PR and comment on stuff you find confusing or which is outright wrong. Weeding out issues early most often leads to a PR being merged faster :)

**Comment by pks-t** [comment]
This has been implemented via #4244
